### ISSUE TITLE
didn't pair sonof rf bridge from telis 4 RTS remote

### ISSUE THREAD

**Issue by fosterym** [sehaas/yfmos#9]
Good day
tasmota 8.1 with Portisch firmware for RF

I sniffed code from rfraw 177 push button from remote
01:52:13 MQT: rf433-bridge/rf433-bridge/RESULT = {"Time":"1970-01-01T01:52:13","RfRaw":{"Data":"AA B1 05 0A1E 12FC 050A 028A 8106 00000000000000122222233333333222233223322223333233233333333233233332332333322333333333333333333333333334 55"}}

and put it to your code:
(yfmos) root@raspberrypi:/home/pi/yfmos# python yfmos.py init --device 0x144051 --rollingcode 0 --profile mainroom2 AA B1 05 0A1E 12FC 050A 028A 8106 00000000000000122222233333333222233223322223333233233333333233233332332333322333333333333333333333333334 55
(init) next: 1, second: True
(yfmos) root@raspberrypi:/home/pi/yfmos# python yfmos.py gen --profile mainroom2 --repeat 8 --command PROG  
Group       A       B       C       D       F               G                    
Byte:       0H      0L      1H      1L      2       3       4       5       6    
        +-------+-------+-------+-------+-------+-------+-------+-------+-------+
        !  0xA  + R-KEY ! C M D + C K S !  Rollingcode  ! Remote Handheld Addr. !
        !  0xA  +  0x1  !  0x8  +  0x7  !    0x0001     !       0x144051        !
        +-------+-------+-------+-------+MSB----+----LSB+LSB----+-------+----MSB+
RfRaw AA B0 3D 05 08 0A1E 12FC 050A 028A 8106 000000000000001222233333322332233233233332233233233332233233332332332332332233332332332332233332234 55

after that, switched working remote to pair mode (shades make a sharp movement)

and try to put command from web-console
RfRaw AA B0 3D 05 08 0A1E 12FC 050A 028A 8106 000000000000001222233333322332233233233332233233233332233233332332332332332233332332332332233332234 55

but shades didn't react:-(
please help to resolve problem

**Comment by dzwiedzmin** [comment]
Try repeat 2, worked for me for the same remote. Recommended repeat 8 didn't.

**Comment by sehaas** [comment]
Is the device ID `0x144051` unique or did you reuse it? If you reuse it, you have to start with rolling code slightly higher than the last one. Is the RF bridge near the shade?
The number of repeats represent the thy of keypress. Low number => short press; Higher number => long keypress.

**Comment by fosterym** [comment]
it's a sonoff rf bridge "Flash Chip Id".  what is it "device"? where I can find it?
rf bridge  in 2 meter from shade

**Comment by fosterym** [comment]
I think I did not understand the procedure correctly.
can you tell the step by step instruction? 
I want to connect the control to the Home Assistant

**Comment by rplevka** [comment]
@fosterym - every time you're retrying the procedure again, make sure you change the device id to a unique new value (it doesn't matter which one, maybe just increment the id by 1). The id is fictional, you're just making your own virtual remote control:
so e.g. in your code, the
`--device 0x144051` should be changed to e.g. `--device 0x144052` on retrying.

btw, here are the step I use:

1. [in tasmota console] `rfraw 177`
2. [on somfy remote] I keep `programming btn` pushed and sniff its code (that saves me 1 step)
3. [on PC] `yfmos.py init --device 0x000001 --rollingcode 0 --profile mainroom2 <last captured string from tasmota console>`
4. [in tasmota console] `backlog rfraw <B1 string generated by yfmos init>`
5. [roller bladed] confirm the registration with sharp movement
6. [on PC] `yfmos run --profile mainroom2 --command UP`

- i see you omitted the `backlog` command, i''m not suer that is the culprit, but perhaps try to use it.

**Comment by sehaas** [comment]
For every new virtual remote you come up with your own, unique device id. You should not reuse the id of your physical remote, otherwise you'll get conflicts with the rolling code.

@rplevka described all required steps - that should work.

It would be nice to have an HA integration. It is on my todo list, but I haven't even started with it.

**Comment by dzwiedzmin** [comment]
Feel free to use my solution for Home Assistant integration, works like a charm:
https://github.com/dzwiedzmin/yfmos/blob/master/yfmosha.py

**Comment by fosterym** [comment]
@rplevka Thank You! I'm succefuly pair rf433 with somfy roller shutter

@dzwiedzmin I tryed to use your script with HA. But it work with error. Partialy rfraw command is broken and didn't apply with rf433

for example:
correct DOWN command: AA B0 43 05 10 09EC 12CA 04F6 0276 6AFE 00000000000000122223333332333333223333333333332233333333332333322333323333333333333333233332233332333322333334 55

incorrect  MY command: AA B0 3B 05 10 09EC 12CA 04F6 0276 6AFE 00000000000000122223333332332233333322332233333322332332233332233222222223322333322332233332234 55

rf433 bridge makes a squeaky sound  and didn't accept MY command

**Comment by rplevka** [comment]
@fosterym that might be this issue:
https://github.com/sehaas/yfmos/issues/11

can you try to execute some command at least 16 times  in a row and see whether some of them get through successfully? (get prepared for a lot of beeping ;))

**Comment by fosterym** [comment]
Yes. I did it. And after 15 times repeat the same command MY (bridge was beep) it do it. But after this the same problem with command UP and DOWN.
I pushed 15 times command UP, after this action UP and DOWN works, but return problem with command MY :-(((

**Comment by dzwiedzmin** [comment]
> Yes. I did it. And after 15 times repeat the same command MY (bridge was beep) it do it. But after this the same problem with command UP and DOWN.
> I pushed 15 times command UP, after this action UP and DOWN works, but return problem with command MY :-(((

What RF bridge are you using? If Sonoff RF Bridge with Tasmota maybe try updating the firmware to the latest ?

**Comment by fosterym** [comment]
sonof RF WIFI bridge with tasmota 8.1.0

---------------
Mishchenko Yury
mischenko.yuri@gmail.com

**Comment by rplevka** [comment]
@fosterym,
you might try to apply my workaround. It's ugly, but it gets the thing done:
https://github.com/rplevka/yfmos/commit/b8776427c866e41e433cb421f6c2228f5aa29598
(tl;dr - it simply keeps generating [and discarding] the B0 string until a valid one (its length is an even number) is generated).

**Comment by fosterym** [comment]
@rplevka 

Sory. But I 'm not  developer:-( can You describe what I need to do?
thx

**Comment by rplevka** [comment]
@fosterym 
just replace your current `yfmos.py` file by this one:
https://raw.githubusercontent.com/rplevka/yfmos/b8776427c866e41e433cb421f6c2228f5aa29598/yfmos.py
(do not forget to back up the original ;) )

**Comment by fosterym** [comment]
Guys, I resolved my case. I bought a russian usb adapter with Somfy and AKKA support and it works perfect througm command line to serial commands.

https://smarttone.ru/shop/rf-blinds-usb/

https://smarttone.ru/blog/home-assistant-rf-blinds/
